src/otx/v2/adapters/torch/mmengine/mmdet/modules/evaluation/evaluator.py

```diff
@@ -0,0 +1,469 @@
+
+    Returns:
+        Tuple[np.ndarray, np.ndarray, float]: tp, fp, miou for each image
+    """
+    num_dets = len(det[0])  # M
+    num_gts = len(gt_masks)  # N
+
+    tp = np.zeros(num_dets, dtype=np.float32)  # pylint: disable=invalid-name
+    fp = np.zeros(num_dets, dtype=np.float32)  # pylint: disable=invalid-name
+    gt_covered_iou = np.zeros(num_gts, dtype=np.float32)
+
+    if len(gt_masks) == 0:
+        fp[...] = 1
+        return tp, fp, 0.0
+    if num_dets == 0:
+        return tp, fp, 0.0
+
+    ious = mask_iou(det, gt_masks, iou_thr)  # (M, N)
+
+    # for each det, the max iou with all gts
+    ious_max = ious.max(axis=1)
+    # for each det, which gt overlaps most with it
+    ious_argmax = ious.argmax(axis=1)
+    # sort all dets in descending order by scores
+    sort_inds = np.argsort(-cls_scores)
+
+    gt_covered = np.zeros(num_gts, dtype=bool)
+    # if no area range is specified, gt_area_ignore is all False
+    for i in sort_inds:
+        if ious_max[i] >= iou_thr:
+            matched_gt = ious_argmax[i]
+            if not gt_covered[matched_gt]:
+                gt_covered[matched_gt] = True
+                gt_covered_iou[matched_gt] = ious_max[i]
+                tp[i] = 1
+            else:
+                fp[i] = 1
+            # otherwise ignore this detected bbox, tp = 0, fp = 0
+        else:
+            fp[i] = 1
+    return tp, fp, np.mean(gt_covered_iou)
+
+
+@METRICS.register_module()
+class OTXDetMetric(VOCMetric):
+    """OTX Object Detection Evaluator for mAP and mIoU."""
+
+    nproc = 4
+
+    def get_gt_instance_masks(self, annotation: list[dict]) -> list[list]:
+        """Format ground truth instance mask annotation.
+
+        Args:
+            annotation (List[Dict]): per-image ground truth annotation
+
+        Returns:
+            cls_anno_list: per-class ground truth instance mask list
+        """
+        cls_anno_list: list[list] = [[] for _ in range(self.num_classes)]
+        for class_id in range(self.num_classes):
+            for ann in annotation:
+                gt_inds = ann["labels"] == class_id
+                polygon_masks = []
+                if gt_inds.any():
+                    gt_inds = np.where(gt_inds == 1)[0]
+                    polygon_masks = ann["masks"][gt_inds]
+                cls_anno_list[class_id].append(polygon_masks)
+        return cls_anno_list
+
+    def get_mask_det_results(self, det_results: list[tuple], class_id: int) -> tuple[list, list]:
+        """Get mask detection results for a specific class.
+
+        Args:
+            det_results (list(tuple)): detection results including bboxes and masks
+            class_id (int): class index
+
+        Returns:
+            cls_dets: per-class detection results including bboxes and decoded masks
+            cls_scores: class scores
+        """
+        cls_scores = [img_res[0][class_id][..., -1] for img_res in det_results]
+        cls_dets: list[tuple] = []
+        for det in det_results:
```

Comment:
```suggestion
        for result in det_results:
```

or alternatively `det_result`